### PR TITLE
cli: add 'tilt alpha updog', a command to experiment with the new api

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas, @maiamcc,

Please review the following commits I made in branch nicks/updog:

36177fc677615813f9cfe854f3c971312cc70f21 (2021-03-16 18:59:18 -0400)
cli: add 'tilt alpha updog', a command to experiment with the new api

a3585fc8262057c2c55104fe87f7296f9fdda2fd (2021-03-16 18:01:57 -0400)
fix

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
Most of this PR is boilerplate from 'tilt up' and 'tilt ci'

File: internal/cli/updog.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	cmdTags := engineanalytics.CmdTags(map[string]string{})
+	a.Incr("cmd.updog", cmdTags.AsMap())
+	defer a.Flush(time.Second)
+
+	deferred := logger.NewDeferredLogger(ctx)
+	ctx = redirectLogs(ctx, deferred)
+
+	log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+
+	// Force wev-mode to prod
```
Comment (maiamcc) [inline internal/cli/updog.go:68]:
```suggestion
	// Force web-mode to prod
```

File: internal/cli/updog.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+func newUpdogCmd() *updogCmd {
+	return &updogCmd{}
+}
+
+func (c *updogCmd) name() model.TiltSubcommand { return "updog" }
+
+func (c *updogCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "updog",
+		Short: "Runs the Tilt apiserver, applies the config, and streams logs until ctrl-c",
```
Comment (maiamcc) [inline internal/cli/updog.go:39]: maybe short doc should mention that this is a dummy app?

**Review by maiamcc — APPROVED** [review]
What's `updog`?

File: internal/cli/updog.go (hunk 2 above)
Comment (nicks) [reply internal/cli/updog.go]: done!

File: internal/cli/updog.go (hunk 1 above)
Comment (nicks) [reply internal/cli/updog.go]: done

**Pull request merged**